### PR TITLE
Add inline documentation

### PR DESCRIPTION
**Pull request by rageagainsthepc**

### CODE REVIEW

File: vmicore/src/include/vmicore/plugins/PluginInterface.h [hunk 1]
```diff
@@ -23,35 +21,100 @@ namespace VmiCore::Plugin
...
+         * Only useful if using a gRPC connection, does nothing otherwise. Will send an error event via a separate
+         * channel which indicates that the run is not successful.
+         */
         virtual void sendErrorEvent(std::string_view message) const = 0;
 
+        /**
+         * Only useful if using a gRPC connection, does nothing otherwise. Will send an inmemory scanner detection event
+         * via a gRPC channel.
+         */
         virtual void sendInMemDetectionEvent(std::string_view message) const = 0;
```
Comment (K-Mayer) [inline vmicore/src/include/vmicore/plugins/PluginInterface.h:110]: nit: Not part of this PR, but I think we should replace this with a more plugin agnostic version.

Comment (rageagainsthepc) [reply vmicore/src/include/vmicore/plugins/PluginInterface.h:112]: Sure, you could create an issue for that.

**Review by K-Mayer — APPROVED** [review]
